src/app/controllers/followerships_controller.rb

```diff
@@ -0,0 +1,17 @@
+class FollowershipsController < ApplicationController
+  before_action :set_user
+
+  def create
+    current_user.follow(@user.id)
+    redirect_to user_functions_path(@user.username)
```

Comment:
It'll be better to redirect_back with fallback, checkout other controllers for an example